How-To-Setup-Internal-Package-Repository-Nexus.md

```diff
@@ -0,0 +1,443 @@
+
+# This will be the URL you need to use as your `--source` parameter and we
+# will call it `<HOSTED REPOSITORY PACKAGE SOURCE URL>` going forward
+"http://localhost:8081/nexus/service/local/nuget/{0}" -f $result.data.id
+
+# The result of that command should be:
+# http://localhost:8081/nexus/service/local/nuget/test-repo
+```
+
+At this stage we have created the repository. Now let's just check we can use it. Run `choco list --source <HOSTED REPOSITORY PACKAGE SOURCE URL>`.
+
+```powershell
+> choco list --source "http://localhost:8081/nexus/service/local/nuget/test-repo/"
+
+Chocolatey v0.10.13
+0 packages found.
+```
+
+There are no packages yet in the repository so we are expecting `0 packages found` so this is a successful test.
+
+### Create User To Push Packages To The Repository
+
+By default anybody will be able to install / upgrade packages from the repository. However only authenticated users will be able to push packages to the repository. The authentication used is an **API Key**. Let's go ahead and create a user with permission to push packages to the repository.
+
+When creating a user to push packages we can again create it through the browser or using PowerShell. If you want to follow the steps and create it through the browser, follow these steps:
+
+1. If you are not already logged in as an admin user:
+   * **Log In** as the `admin` user (or another user you have created with Administrator privileges);
+   * If you have not changed the defaults then see [**Step 1: Change the Administrative Password and Email Address**](https://help.sonatype.com/repomanager3/installation/post-install-checklist) for the defaults. **If the repository will be used in production we strongly recommend you change your username and / or password.**;
+1. Go to **Security** -> **Users** and then click the **Add** button and then **Nexus User** (note if you hover over the `?` at the end of each field it will provide you with help):
+   ![Create User to Push Packages in Nexus](images/nexus/create-push-user.png)
+   * Enter **User ID** as `push-user`;
+   * Enter **First Name**, **Last Name** and **Email** - anything can be in these fields;
+   * Set **Status** to `Active`
+   * Set **New Password** and **Confirm Password** to `password` - **you should use a secure password in a production environment**;
+   * When you get to the **Role Management** field, click the **Add** button and add the following roles:
+     * `Nexus Api-Key Access` - this allows the user to authenticate using the NuGet API Key (unless you have setup an alternative method of authentication you will want to use this);
+     * `UI: Base UI Privileges` and `UI: Repository Browser` - this will allow the user to log into the Nexus Dashboard and retrieve their API Key. Once this is done you can remove these roles if you do not want the user to be able to log in;
+     * `Repo: All NuGet Repositories (Full Control)` - this access allows the user to push packages to the repository;

```

Comment:
This would be a great place to mention that if none of this is seen, they need to add the NuGet API realm in security settings and point them back to previous steps.